alibi/tests/test_saving.py

```diff
@@ -334,3 +404,34 @@ def test_save_TreeShap(tree_explainer, rf_classifier, iris_data):
 
         # TreeShap is deterministic
         assert_allclose(exp0.shap_values[0], exp1.shap_values[0])
+
+
+@pytest.mark.parametrize('rf_classifier', [lazy_fixture('iris_data')], indirect=True)
+def test_save_cfrl(cfrl_explainer, rf_classifier, iris_data):
+    X = iris_data['X_test']
+    exp0 = cfrl_explainer.explain(X=X, Y_t=np.array([0]))
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # save explainer
+        cfrl_explainer.save(temp_dir)
+
+        # define predictor and load the explainer
+        predictor = lambda x: rf_classifier[0].predict_proba(x)  # noqa: E731
+        cfrl_explainer1 = load_explainer(temp_dir, predictor=predictor)
+        assert isinstance(cfrl_explainer1, CounterfactualRLTabular)
+
+        # check metadata. loading can change the class, so we have to remove the encoder, decoder, actor and critic
+        # see: https://www.tensorflow.org/api_docs/python/tf/saved_model/load for more details
+        # also have to remove the optimizers since those are not saved.
+        keys_to_remove = ['encoder', 'decoder', 'actor', 'critic', 'optimizer_actor', 'optimizer_critic', 'callbacks']
+        for key in keys_to_remove:
+            cfrl_explainer.meta["params"].pop(key)
+            cfrl_explainer1.meta["params"].pop(key)
```

Comment:
Not quite sure I follow why all those are missing, is it just because we don't save them? But surely all those models are required at prediction time...?